network/message/authorization.go

```diff
@@ -0,0 +1,375 @@
+func (m MsgAuthConfig) IsAuthorized(role flow.Role, channel channels.Channel) error {
+	authorizedRoles, ok := m.Config[channel]
+	if !ok {
+		return ErrUnauthorizedMessageOnChannel
+	}
+
+	if !authorizedRoles.Contains(role) {
+		return ErrUnauthorizedRole
+	}
+
+	return nil
+}
+
+var (
+	ErrUnknownMsgType               = errors.New("could not get authorization config for unknown message type")
+	ErrUnauthorizedMessageOnChannel = errors.New("message is not authorized to be sent on channel")
+	ErrUnauthorizedRole             = errors.New("sender role not authorized to send message on channel")
+	AuthorizationConfigs            map[string]MsgAuthConfig
+)
+
+func initializeMessageAuthConfigsMap() {
+	AuthorizationConfigs = make(map[string]MsgAuthConfig)
+
+	// consensus
+	AuthorizationConfigs[BlockProposal] = MsgAuthConfig{
+		Name: BlockProposal,
+		Type: func() interface{} {
+			return new(messages.BlockProposal)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ConsensusCommittee: {flow.RoleConsensus},
+			channels.PushBlocks:         {flow.RoleConsensus}, // channel alias ReceiveBlocks = PushBlocks
+		},
+	}
+	AuthorizationConfigs[BlockVote] = MsgAuthConfig{
+		Name: BlockVote,
+		Type: func() interface{} {
+			return new(messages.BlockVote)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ConsensusCommittee: {flow.RoleConsensus},
+		},
+	}
+
+	// protocol state sync
+	AuthorizationConfigs[SyncRequest] = MsgAuthConfig{
+		Name: SyncRequest,
+		Type: func() interface{} {
+			return new(messages.SyncRequest)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.SyncCommittee:     flow.Roles(),
+			channels.SyncClusterPrefix: {flow.RoleCollection},
+		},
+	}
+	AuthorizationConfigs[SyncResponse] = MsgAuthConfig{
+		Name: SyncResponse,
+		Type: func() interface{} {
+			return new(messages.SyncResponse)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.SyncCommittee:     flow.Roles(),
+			channels.SyncClusterPrefix: {flow.RoleCollection},
+		},
+	}
+	AuthorizationConfigs[RangeRequest] = MsgAuthConfig{
+		Name: RangeRequest,
+		Type: func() interface{} {
+			return new(messages.RangeRequest)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.SyncCommittee:     flow.Roles(),
+			channels.SyncClusterPrefix: {flow.RoleCollection},
+		},
+	}
+	AuthorizationConfigs[BatchRequest] = MsgAuthConfig{
+		Name: BatchRequest,
+		Type: func() interface{} {
+			return new(messages.BatchRequest)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.SyncCommittee:     flow.Roles(),
+			channels.SyncClusterPrefix: {flow.RoleCollection},
+		},
+	}
+	AuthorizationConfigs[BlockResponse] = MsgAuthConfig{
+		Name: BlockResponse,
+		Type: func() interface{} {
+			return new(messages.BlockResponse)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.SyncCommittee:     flow.Roles(),
```

Comment:
This should be just consensus nodes:
https://github.com/onflow/flow-go/blob/master/cmd/scaffold.go#L696-L696
```suggestion
			channels.SyncCommittee:     {flow.RoleConsensus},
```